src/geometry/regionbasedadjacency.cpp

```diff
@@ -0,0 +1,44 @@
+#include <libgeodecomp/geometry/regionbasedadjacency.h>
+#include <libgeodecomp/geometry/coordbox.h>
+#include <libgeodecomp/geometry/region.h>
+
+namespace LibGeoDecomp
+{
+
+RegionBasedAdjacency::RegionBasedAdjacency() :
+    region(new Region<2>())
+{
+
+}
+
+void RegionBasedAdjacency::insert(int from, int to) {
+    (*region) << Coord<2>(to, from);
```

Comment:
D'oh. Sorry, I got it. OK, I'll make adjacency's type a template argument to Region::expandWithAdjacency(), that will break the cyclic inclusion.